### PR TITLE
8278353: Provide Duke as default favicon in Simple Web Server

### PR DESCRIPTION
**Pull request by sormuras**
Please pre-review this draft pull request adding a default for response for the `/favicon.ico` path to the Simple Web Server introduced by [JEP 408](https://openjdk.org/jeps/408). The image in this PR is a copy of https://inside.java/images/favicon.ico

Before this change, an initial request from a browser yields the following output on the server console:
```
Binding to loopback by default. For all interfaces use "-b 0.0.0.0" or "-b ::".
Serving PATH and subdirectories on 127.0.0.1 port 8000
URL http://127.0.0.1:8000/
127.0.0.1 - - [30/Apr/2024:13:11:39 +0200] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [30/Apr/2024:13:11:40 +0200] "GET /favicon.ico HTTP/1.1" 404 -
```

After applying this change set, the two last line read:
```
127.0.0.1 - - [30/Apr/2024:13:12:08 +0200] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [30/Apr/2024:13:12:08 +0200] "GET /favicon.ico HTTP/1.1" 200 -
```

Note that if the directory being served contains a `favicon.ico` file it is still sent in favor of the default icon introduced by this change.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8278353](https://bugs.openjdk.org/browse/JDK-8278353): Provide Duke as default favicon in Simple Web Server (**Enhancement** - P4)

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19020/head:pull/19020` \
`$ git checkout pull/19020`

Update a local copy of the PR: \
`$ git checkout pull/19020` \
`$ git pull https://git.openjdk.org/jdk.git pull/19020/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19020`

View PR using the GUI difftool: \
`$ git pr show -t 19020`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19020.diff">https://git.openjdk.org/jdk/pull/19020.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19020#issuecomment-2097536322)

### CODE REVIEW

File: src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/FileServerHandler.java [hunk 1]
```diff
@@ -250,6 +254,36 @@ private static Path indexFile(Path path) {
...
+
+    private void serveDefaultFavIcon(HttpExchange exchange, boolean writeBody)
+            throws IOException
+    {
+        var respHdrs = exchange.getResponseHeaders();
+        var icon = "/sun/net/httpserver/simpleserver/resources/favicon.ico";
+        try (var stream = getClass().getModule().getResourceAsStream(icon)) {
+            var bytes = stream.readAllBytes();
+            respHdrs.set("Content-Type", "image/x-icon");
+            // TODO respHdrs.set("Last-Modified", getLastModified(...));
```
Comment (sormuras) [inline src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/FileServerHandler.java:269]: Is there a canonical way to retrieve a stable timestamp for the built-in `favicon.ico` here?

Comment (sormuras) [reply src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/FileServerHandler.java]: Using Java's birthday `"Mon, 23 May 1995 11:11:11 GMT"` for the time being.

Comment (djelinski) [reply src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/FileServerHandler.java]: [java.version.date](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/System.html#java.version.date) could be a reasonable alternative

Comment (sormuras) [reply src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/FileServerHandler.java]: I thought about that too, but dates from the future might lead to other issues like caching.

`java -XshowSettings:properties --version`
```
...
java.version.date = 2024-09-17
...
openjdk 23-ea 2024-09-17
OpenJDK Runtime Environment (build 23-ea+21-1723)
OpenJDK 64-Bit Server VM (build 23-ea+21-1723, mixed mode, sharing)
```

**Review by dfuch — APPROVED** [review]
LGTM

**Pull request closed without merging**